packages/layers/src/multiscale-image-layer/multiscale-image-layer.js

```diff
@@ -99,11 +128,22 @@ const MultiscaleImageLayer = class extends CompositeLayer {
          * return type, and optional throw for performance.
          */
         const tiles = await Promise.all(selections.map(getTile));
-
         const tile = {
-          data: tiles.map(d => d.data),
-          width: tiles[0].width,
-          height: tiles[0].height
+          data: tiles.map(d =>
+            clip({
+              data: d.data,
+              width: tiles[0].width,
+              height: tiles[0].height
+            })
+          ),
+          width:
+            isWidthUnderTileSize && tiles[0].height === tileSize
+              ? clippedWidth
+              : tiles[0].width,
+          height:
+            isHeightUnderTileSize && tiles[0].height === tileSize
+              ? clippedHeight
+              : tiles[0].height
```

Comment:
IIRC, I seem to remember adding padding to tiles from the loaders. It seems like we are just reversing that here? Perhaps this warrants a change to the loaders?

Reply:
So this was also my first instinct but I don't think we do anymore after you found that specific WebGL setting for controlling of what number the textures should be a multiple, which [we set to 1](https://github.com/hms-dbmi/viv/blob/master/packages/layers/src/xr-layer/xr-layer.js#L144-L148).